src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -33,22 +36,82 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public static class ThreadUtility
     {
+        private const int ThreadAbortedCheckDelay = 100;
+
+
+        /// <summary>
+        /// Pre-Task compatibility
+        /// </summary>
+        public static void Delay(int milliseconds, WaitCallback threadPoolWork, object state = null)
+        {
+            new DelayClosure(milliseconds, threadPoolWork, state);
+        }
+
+        private sealed class DelayClosure
+        {
+            private readonly Timer _timer;
+            private readonly WaitCallback _threadPoolWork;
+            private readonly object _state;
+
+            public DelayClosure(int milliseconds, WaitCallback threadPoolWork, object state)
+            {
+                _threadPoolWork = threadPoolWork;
+                _state = state;
+                _timer = new Timer(Callback, null, Timeout.Infinite, Timeout.Infinite);
+                // Make sure _timer is set before starting the timer so that it is disposed.
```

Comment:
@rprouse Only in this case. As I was writing the code I visualized this sequence of events:

 1. The expression `new Timer(Callback, null, milliseconds, Timeout.Infinite)` is evaluated
 2. `milliseconds` is so small that the callback is called before step 3. You get a null reference exception in `Callback` because the `_timer field is not yet set.
 3. The `_timer` field is set to the value returned from evaluating the` new Timer` expression.